### PR TITLE
Fix guardicore

### PR DESCRIPTION
**Pull request by AradCarmi**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/XSUP-20878)

## Description
Fixed an issue where the **fetch-incidents** command failed to fetch incidents.

### CODE REVIEW

File: Packs/GuardiCore/ReleaseNotes/2_0_8.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+
+#### Integrations
+
+##### GuardiCore v2
+- Updated the Docker image to: *demisto/python3:3.10.10.48392*.
+
+- Fixed an issue where the **fetch-incidents** command failed to fetch incidents.
```
Comment (dantavori) [inline Packs/GuardiCore/ReleaseNotes/2_0_8.md:7]:
```suggestion
- Updated the Docker image to: *demisto/python3:3.10.10.48392*.
- Fixed an issue where the ***fetch-incidents*** command failed to fetch new incidents.
```

File: Packs/GuardiCore/Integrations/GuardiCoreV2/GuardiCoreV2_test.py [hunk 2]
```diff
@@ -416,3 +416,46 @@ def test_fetch_incidents(mocker, requests_mock):
...
+    requests_mock.post(
+        'https://api.guardicoreexample.com/api/v3.0/authenticate',
+        json={'access_token': TEST_API_KEY})
+    requests_mock.get('https://api.guardicoreexample.com/api/v3.0/incidents',
+                      json=incidents_data.get('second'))
+
+    client = Client(base_url='https://api.guardicoreexample.com/api/v3.0',
+                    verify=False, proxy=False, username='test', password='test')
+    incidents, last_fetch = fetch_incidents(client, {
+        'first_fetch': '40 years'})  # if xsoar is still here when this is a bug then we have a good problem on our hands :)
```
Comment (dantavori) [inline Packs/GuardiCore/Integrations/GuardiCoreV2/GuardiCoreV2_test.py:446]: any particular reason for this value?

File: Packs/GuardiCore/Integrations/GuardiCoreV2/GuardiCoreV2.py [hunk 3]
```diff
@@ -333,18 +333,21 @@ def fetch_incidents(client: Client, args: Dict[str, Any]) -> \
...
-            'occurred': timestamp_to_datestring(start_time, DATE_FORMAT),
-            'severity': incident_severity_to_dbot_score(severity),
-            'rawJSON': json.dumps(inc)
-        }
-        incidents.append(incident)
-
+        demisto.debug(
+            f"Fetch incident checking id: {id} with start time of:"
+            f" {start_time}. LastRun time was: {current_fetch}")
         if current_fetch < start_time:
```
Comment (dantavori) [inline Packs/GuardiCore/Integrations/GuardiCoreV2/GuardiCoreV2.py:339]: is it guaranteed that the results are retrieved in ascending order?
e.g. if current_fetch = 10 and we got two incidents, the first has current_time=12 and the second has current_time=11 - you won't fetch the second one.
please check this, and if needed, either always use the first value of current_fetch, or sort the results.

File: Packs/GuardiCore/ReleaseNotes/2_0_8.md (hunk 1 above)
Comment (AradCarmi) [reply Packs/GuardiCore/ReleaseNotes/2_0_8.md:7]: There is an issue with the update-release-notes, already opened a ticket.

File: Packs/GuardiCore/Integrations/GuardiCoreV2/GuardiCoreV2_test.py (hunk 2 above)
Comment (AradCarmi) [reply Packs/GuardiCore/Integrations/GuardiCoreV2/GuardiCoreV2_test.py:446]: Not really, just copied a working test

File: Packs/GuardiCore/Integrations/GuardiCoreV2/GuardiCoreV2.py (hunk 3 above)
Comment (AradCarmi) [reply Packs/GuardiCore/Integrations/GuardiCoreV2/GuardiCoreV2.py:339]: The customer approved the fix but we can add `{sort: descending}` to the fetch params. (but we don't have an instance to check it out)

**Review by dantavori — APPROVED** [review]

**Pull request merged**